src/lib/src/service/execution/test.execution.service.ts

```diff
@@ -1,48 +1,72 @@
+export class DefaultTestExecutionService extends TestExecutionService {
 
   private static readonly statusURLPath = '/status';
   private static readonly executeURLPath = '/execute';
   private static readonly statusAllURLPath = '/status/all';
   private serviceUrl: string;
 
-  constructor(private http: AuthHttp, config: TestExecutionServiceConfig) {
+  constructor(private http: HttpClient, config: TestExecutionServiceConfig) {
+    super();
     this.serviceUrl = config.testExecutionServiceUrl;
   }
 
-  execute(path: string): Promise<Response> {
-    return this.http.post(this.getURL(path, TestExecutionService.executeURLPath), '').toPromise();
+  execute(path: string): Promise<any> {
+    return this.http.post(this.getURL(path, DefaultTestExecutionService.executeURLPath), '').toPromise();
   }
 
-  status(path: string): Promise<Response> {
-    return this.http.get(this.getURL(path, TestExecutionService.statusURLPath) + '&wait=true').toPromise();
+  getStatus(path: string): Promise<TestExecutionStatus> {
+    return this.http.get(this.getURL(path, DefaultTestExecutionService.statusURLPath) + '&wait=true', { responseType: 'text' })
+      .toPromise().then(text => {
+        const status: TestExecutionStatus = { path: path, status: this.toTestExecutionState(text) };
+        return status;
+      });
   }
 
-  statusAll(): Promise<Map<string, ElementState>> {
-    return this.http.get(`${this.serviceUrl}${TestExecutionService.statusAllURLPath}`).toPromise().then(response => {
-      let statusUpdates: TestStatusInfo[] = response.json();
-      return new Map(statusUpdates.map(update => [update.path, this.toElementState(update.status)] as [string, ElementState]));
+  getAllStatus(): Promise<TestExecutionStatus[]> {
+    return this.http.get<any[]>(`${this.serviceUrl}${DefaultTestExecutionService.statusAllURLPath}`).toPromise().then(statusResponse => {
+      const status: any[] = statusResponse;
+      status.forEach((value) => { value.status = this.toTestExecutionState(value.status); });
+      return status;
     });
   }
 
   private getURL(workspaceElementPath: string, urlPath: string = ''): string {
-    let encodedPath = workspaceElementPath.split('/').map(encodeURIComponent).join('/');
+    const encodedPath = workspaceElementPath.split('/').map(encodeURIComponent).join('/');
     return `${this.serviceUrl}${urlPath}?resource=${encodedPath}`;
   }
 
-  private toElementState(status: string): ElementState {
-    switch (status) {
-      case 'RUNNING': return ElementState.Running;
-      case 'SUCCESS': return ElementState.LastRunSuccessful;
-      case 'FAILED': return ElementState.LastRunFailed;
+  private toTestExecutionState(state: string): TestExecutionState {
+    switch (state) {
+      case 'RUNNING': return TestExecutionState.Running;
+      case 'FAILED': return TestExecutionState.LastRunFailed;
+      case 'SUCCESS': return TestExecutionState.LastRunSuccessful;
+      case 'IDLE': return TestExecutionState.Idle;
```

Comment:
could be abbreviated to 
```
      case 'IDLE':
      default: return TestExecutionState.Idle;
```
(tslint's _no-switch-case-fall-through_ should not complain, either, as long as the case that "falls through" is empty.)